ControlledChaos0/Bloodelic
# Npc vision observer

Added subscriber/publisher model for NPC vision and generalized DetectEntitySight.

To test:

Try DetectEntitySight with several different entities including the monster.

run AddOccupant with different entities:
Other NPCs should be ignored
The Monster should instantly raise suspicion to full
Anything else raises suspicion.

Also note that I could not test that this script actually ignores other NPCs because I used a scene with only one NPC that does not move.

Testing should be done under the MainIntegration scene in my folder.
